libminifi/include/core/Repository.h

```diff
@@ -85,6 +85,10 @@ class Repository : public virtual core::SerializableComponent, public core::Trac
 
   virtual void flush();
 
+  virtual void printStats() {
+    return;
+  }
+
```

Comment:
Introducing an empty function to a base class is a smell. In this case, `printStats` doesn't need to be `virtual` or even present in `Repository`.

From the standpoint of contracts:
By introducing `printStats` to `Repository`, `Repository` now looks like something that can print stats of itself. The reality is that only `ProvenanceRepository` (and potentially later a select other few derived classes) can offer this functionality. In all other cases, this is a broken promise.

From the OOP standpoint:
`printStats` is a method specific to `ProvenanceRepository` but has leaked to `Repository`, breaking encapsulation.

From a performance standpoint:
Why pay for virtual dispatch when in all of the callers of `printStats`, the static and the dynamic types of the object match? We could even go as far as removing the function declaration from the headers altogether and giving it `static` linkage, therefore giving the compiler the freedom to throw the implementation away (except for 1 inlined instance) to reduce code bloat.

Reply:
Wanted to remove and follow up in https://issues.apache.org/jira/browse/MINIFICPP-1145.

Done.